lib/Runtime/Library/ArrayBuffer.cpp

```diff
@@ -148,20 +148,24 @@ namespace Js
     uint32 ArrayBuffer::GetByteLengthFromVar(ScriptContext* scriptContext, Var length)
     {
         Var firstArgument = length;
+        double returnValue;
+
         if (TaggedInt::Is(firstArgument))
         {
-            int32 byteCount = TaggedInt::ToInt32(firstArgument);
-            if (byteCount < 0)
-            {
-                JavascriptError::ThrowRangeError(
-                    scriptContext, JSERR_ArrayLengthConstructIncorrect);
-            }
-            return byteCount;
+            returnValue = TaggedInt::ToInt32(firstArgument);
         }
         else
         {
-            return JavascriptConversion::ToUInt32(firstArgument, scriptContext);
+            returnValue = JavascriptConversion::ToNumber(firstArgument, scriptContext);
         }
+
+        if (returnValue < 0 || std::isinf(returnValue))
```

Comment:
@ianwjhalliday I think there is a bug with `NumberUtilities::IsFinite` as it fails to identify `double(inf)` is not finite.